certified-connectors/DocuSignDemo/script.csx

```diff
@@ -5825,6 +5856,81 @@ private void RenameSpecificKeys(JObject jObject, Dictionary<string, string> keyM
+      var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
+      JObject newBody = new JObject();
+
+      int top = string.IsNullOrEmpty(query.Get("top")) ? 10: int.Parse(query.Get("top"));
+      int skip = string.IsNullOrEmpty(query.Get("skip")) ? 0: int.Parse(query.Get("skip"));
+
+      JArray envelopes = (body["envelopes"] as JArray) ?? new JArray();
+      JArray filteredEnvelopes = new JArray();
+      var filteredEnvelopesDetails = new JArray();
+      var recipientName = query.Get("recipientName") ?? null;
+      var recipientEmailId = query.Get("recipientEmailId") ?? null;
+      var envelopeTitle = query.Get("envelopeTitle") ?? null;
+      var customFieldName = query.Get("customFieldName") ?? null;
+      var customFieldValue = query.Get("customFieldValue") ?? null;
+
+      var envelopeFilterMap = new Dictionary<string, string>() {
+        {"recipientName", recipientName},
+        {"recipientEmailId", recipientEmailId},
+        {"envelopeTitle", envelopeTitle},
+        {"customFieldName", customFieldName},
+        {"customFieldValue", customFieldValue}
+      };
+
+      foreach (var filter in envelopeFilterMap.Keys) 
+      {
+        if (envelopeFilterMap[filter] != null)
+        {
+          switch (filter)
+          {
+            case "recipientName":
+            case "recipientEmailId":
+              filteredEnvelopes = new JArray(envelopes.Where(envelope =>
+                envelope["recipients"].ToString().ToLower().Contains(envelopeFilterMap[filter].ToString().ToLower())));
+              break;
+            case "envelopeTitle":
+              filteredEnvelopes = new JArray(envelopes.Where(envelope =>
+                envelope["emailSubject"].ToString().ToLower().Contains(envelopeFilterMap[filter].ToString().ToLower())));
+              break;
+            case "customFieldName":
+            case "customFieldValue":
+              filteredEnvelopes = new JArray(envelopes.Where(envelope =>
+              {
+                var customFields = envelope["customFields"] as JToken;
+                return customFields?.ToString().ToLower().Contains(envelopeFilterMap[filter].ToString().ToLower()) ?? false;
+              }));
+              break;
+            default:
+              break;
+          }
+
+          if (filteredEnvelopes.Count > 0)
+          {
+            envelopes.Clear();
+            envelopes = new JArray(filteredEnvelopes);
+            filteredEnvelopes.Clear();
+          }
+          else
+          {
+            envelopes.Clear();
+            break;
+          }
+        }
+      }
+
+      filteredEnvelopesDetails = GetFilteredEnvelopeDetails(envelopes);
+      newBody["value"] = (filteredEnvelopesDetails.Count < top) ? 
+        filteredEnvelopesDetails : 
+        new JArray(filteredEnvelopesDetails.Skip(skip).Take(top).ToArray());
+      newBody["hasMoreResults"] = (skip + top < filteredEnvelopesDetails.Count) ? true : false;
```

Comment:
Help me understand the logic, what is the use of "hasMoreResults"?

Reply:
So when you get a list of envelopes, you get about 1000 at a time. If there are more than 1000 envelopes in your query, hasMoreResults will say true, and you can follow up with a subsequent call.

Reply:
Nice. Got it.